engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -287,9 +287,13 @@ namespace dmGameObject
 
     void DeallocCollection(Collection* collection)
     {
+        DM_PROFILE(GameObject, "DeallocCollection");
+
         HRegister regist = collection->m_Register;
         for (uint32_t i = 0; i < regist->m_ComponentTypeCount; ++i)
         {
+            DM_PROFILE_DYN(GameObjectDeleteWorld, regist->m_ComponentTypes[i].m_Name, regist->m_ComponentTypes[i].m_NameHash);
```

Comment:
Make it visible that the deallocation for different component types actually take time.